math-scala/src/main/scala/org/apache/mahout/visualization/MahoutPlot.scala

```diff
@@ -18,7 +18,36 @@
 
 package org.apache.mahout.visualization
 
+import java.awt.Graphics2D
+import java.awt.image.BufferedImage
+import java.io.File
+import javax.imageio.ImageIO
+import javax.swing.JFrame
 
-trait mlpot  {
+import org.apache.mahout.math.Matrix
+import smile.plot.{PlotCanvas, PlotPanel}
+
+
+trait MahoutPlot  {
+
+  var canvas : PlotCanvas = _
+  var plotPanel: PlotPanel = _
+  var plotFrame: JFrame = _
+  var mPlotMatrix: Matrix = _
+  def contentPane = canvas
+
+  // export a PNG of the plot to /tmp/test.png
+  def exportPNG(path: String ="/tmp/test.png") = {
+    val bi: BufferedImage =
+      new BufferedImage(contentPane.getWidth, contentPane.getHeight, BufferedImage.TYPE_INT_ARGB)
+
+    val g2d: Graphics2D = bi.createGraphics
+
+    contentPane.printAll(g2d)
+
+    val file: File = new File(path)
+
+    ImageIO.write(bi, "PNG", file)
+  }
```

Comment:
@pferrel Setup to easily export plot PNGs to the Filesystem.  Maybe we can we can pipe these to the Zeppelin server via JSON?